File: kbe/src/client/unity3d/Property.cs
namespace KBEngine
{
  	using UnityEngine; 
	using System; 
	
    public class Property 
    {
		public string name = "";
    	public KBEDATATYPE_BASE utype = null;
		public UInt16 properUtype = 0;
		public Int16 aliasID = -1;
		
		public string defaultValStr = "";
		public System.Reflection.MethodInfo setmethod = null;
		
		public object val = null;
		
		public Property()
		{
			
		}
    }
    
} 

File: kbe/res/sdk_templates/client/ue4/Source/KBEnginePlugins/Engine/Property.cpp

#include "Property.h"
#include "KBDebug.h"

namespace KBEngine
{

Property::Property():
	name(TEXT("")),
	pUtype(NULL),
	properUtype(0),
	properFlags(0),
	aliasID(-1),
	defaultValStr(TEXT("")),
	pDefaultVal(NULL)
{
}

Property::~Property()
{
	KBE_SAFE_RELEASE(pDefaultVal);
}

}